### ISSUE TITLE
LIMIT_EXCEEDED (429) error when inviting user to a room on a (deep) hierarchical space structure.

### ISSUE THREAD

**Issue by salomaat** [matrix-org/synapse#15882]
### Description

When creating a space hierarchy of several levels and then inviting a user into a room in the deep hierarchy a LIMIT_EXCEEDED (429) arises. This is on single room / single invite not in multi invite on multiple rooms.  

### Steps to reproduce

1 Create two users, 
2 then create a space hierarchy with first user down to 5 levels. 
3 Then create a room into the 5th level of spaces. 
4 Then try invite 2nd user into the room. 
5 You will most likely get 429 error. 

Using standard rc_limits i.e. fresh Synapse setup.

### Homeserver

Another home server running on UpCloud VM 

### Synapse Version

v1.87

### Installation Method

Debian packages from packages.matrix.org

### Database

PostgreSQL

### Workers

Single process

### Platform

Distributor ID:	Debian
Description:	Debian GNU/Linux 10 (buster)
Release:	10
Codename:	buster

Server running on VM in UpCloud

### Configuration

Fresh Synapse install, standard setups

### Relevant log output

```shell
Homeserver.log

2023-07-05 07:30:41,571 - synapse.http.server - 117 - INFO - POST-1016 - <XForwardedForRequest at 0x7fb7c011d4a8 method='POST' uri='/_matrix/client/r0/rooms/!pTDaCacJgHLIhGVQWK%3Adev.eu.taxonomy.app/invite' clientproto='HTTP/1.0' site='8008'> SynapseError: 429 - Too Many Requests
2023-07-05 07:30:41,572 - synapse.access.http.8008 - 482 - INFO - POST-1016 - 94.31.96.232
```

### Anything else that would be useful to know?

-

**Comment by MadLittleMods** [comment]
Possibly related to https://github.com/matrix-org/synapse/issues/14180

**Comment by H-Shay** [comment]
Hmm I wrote a test to check this behavior and it is not resulting in a 429. Steps of the test are:
1. create a user, log them in and create a space (this user creates all subsequent rooms and sub-spaces)
2. create a subspace, add this as a child of the first space
3. create another subspace, add this as a child of the first subspace
4. create another subspace, add this as a child of the second subspace
5. create another subspace, add this as a child of the third subspace
6. create another subspace, add this as a child of the fourth subspace
7. create another subspace, add this as a child of the fifth subspace
8. create a room, add it as a child of the fifth subspace
9. invite a second user to the room created in step 8

This results in a 200 rather than a 403, so I wonder what's happening here. Does this look like the correct procedure to produce this error? Is there anything I am missing here? Does changing the invite ratelimits do anything?

**Comment by H-Shay** [comment]
Also I am assuming both users are local to the homeserver?

**Comment by salomaat** [comment]
> Also I am assuming both users are local to the homeserver?

Correct, invite is not over federation.

**Comment by salomaat** [comment]
> Possibly related to #14180

Agree, this issue #14180 looks similar and/or could share the same root cause.

**Comment by H-Shay** [comment]
I would say the next step would be to turn on DEBUG logging and see if that surfaces anything - I do find it a little odd that the behavior isn't reproducible in a test.

**Comment by salomaat** [comment]
We were able to get pass this suspected bug by setting the following setups on our Synapse. 

rc_joins:
  local:
    per_second: 10
    burst_count: 2000
  remote:
    per_second: 10
    burst_count: 2000

rc_joins_per_room:
  per_second: 10
  burst_count: 2000

rc_invites:
  per_room:
      per_second: 10
      burst_count: 2000
  per_user:
      per_second: 10
      burst_count: 2000
  per_issuer:
      per_second: 10
      burst_count: 2000

rc_login:
  address:
    per_second: 10
    burst_count: 2000
  account:
    per_second: 10
    burst_count: 2000
  failed_attempts:
    per_second: 10
    burst_count: 2000

**Comment by leonardorortizm** [comment]
Same problem

An error occurred connecting to your home server, please try again later.(M_LIMIT_EXCEEDED) when trying to login via element app.
